test/functions/arrow-expression.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright 2014 Shape Security, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var Shift = require("shift-ast");
+
+var testEsprimaEquiv = require('../assertions').testEsprimaEquiv;
+var testParse = require('../assertions').testParse;
+var expr = require("../helpers").expr;
+var testParseFailure = require('../assertions').testParseFailure;
+
+suite("Parser", function () {
+  suite("arrow expression", function () {
+    testEsprimaEquiv("(a) + (b)");
+    testEsprimaEquiv("(a)");
+    testEsprimaEquiv("((a))");
+    testEsprimaEquiv("((a))()");
+    testEsprimaEquiv("((a))((a))");
+    testEsprimaEquiv("void (a)");
+    testEsprimaEquiv("(void a)");
+    testEsprimaEquiv("(a++)");
+    testEsprimaEquiv("(a)++");
+    testEsprimaEquiv("(a)--");
+    testEsprimaEquiv("(a) ? (b) : (c)");
```

Comment:
We really shouldn't be adding new esprima-equiv tests. We're trying to cut these out eventually.